gui/default/syncthing/core/syncthingController.js

```diff
@@ -923,14 +923,14 @@ angular.module('syncthing.core')
             return '';
         };
 
-        $scope.findDeviceFromShort = function (shortID) {
+        $scope.friendlyNameFromShort = function (shortID) {
             var matches = $scope.devices.filter(function (n) {
                 return n.deviceID.substr(0, 7) === shortID;
             });
             if (matches.length !== 1) {
-                return undefined;
+                return "undefined";
             }
-            return matches[0];
+            return matches[0].name;
         };
```

Comment:
No, modifiedBy in Go code is an integer equal to 0, which when serialized to JSON becomes AAAAA (and the snippet @calmh posted changes that to be an empty string instead, which should be part of this PR).

We said that we want to:

1. Display device name if possible
2. Display short ID (if showing device name is not possible as we are not directly connected to the device)
3. Show 'Unknown' if it's a modification by a legacy client which hasn't set the field (resulting in AAAAA now and empty string once you apply calmh's snippet)